### ISSUE TITLE
Prediction on left and right white matter

### ISSUE THREAD

**Issue by chocolatetoast-chiu** [MIC-DKFZ/nnUNet#1613]
Hello!

First of all, thank you and your team for making this powerful segmentation model network. I can now easily set up my model. But I wonder if it is possible to make the model distinguish the left and right parts of cerebral white matter. 

I asked this because I trained my model by choosing 3d fullres with almost 3000 training sets with 110 epochs. I cannot easily distinguish the complete left white matter and white matter. (Please refer to the attached screenshots)

Ground truth
![image](https://github.com/MIC-DKFZ/nnUNet/assets/56018137/8c83074f-9c7b-474e-a4d3-fdeb051874a6)
Predicted result
![image](https://github.com/MIC-DKFZ/nnUNet/assets/56018137/641210aa-5ef9-459b-bf59-e85f453b91f2)

I'm curious if increasing the epoch could lead to better results. Or, is there anything else I could try? Thank you!

BTW, I notice there is a "NoMirror augmentation" setting in V1, and it looks like it disappears. So I made left and right labels into the same segmentation ground truth referring to this issue [#80](https://github.com/MIC-DKFZ/nnUNet/issues/80)

**Comment by chocolatetoast-chiu** [comment]
Here is an update on my end.

I found that there is an option that decides whether to do mirroring or not in this section in nnUNetTrainer.py.
```python
 if mirror_axes is not None and len(mirror_axes) > 0:
            tr_transforms.append(MirrorTransform(mirror_axes))
```

After I commented this section, the training log showed better improvement:
![image](https://github.com/MIC-DKFZ/nnUNet/assets/56018137/0dfa7044-6798-4753-9f9c-05c9f1d08e5a)

I wonder if this is the right step to do to deal with the left and right labels on similar structures in the brain. Thanks!


Best,

**Comment by lisakausch** [comment]
Hi @chocolatetoast-chiu ,

there is the trainer `nnUNetTrainerNoMirroring`, you can use it by passing this trainer as argument to `nnUNetv2_train`

Best,
Lisa

**Comment by chocolatetoast-chiu** [comment]
Hi @lisakausch ,


Thanks for pointing it out! I didn't notice that! But that may be helpful. SinceI'm facing a challenge right after I commented on the section I mentioned before:
```python
if mirror_axes is not None and len(mirror_axes) > 0:
            tr_transforms.append(MirrorTransform(mirror_axes))
```

The training log showed that I got a pretty good pseudo-Dice score on the left and right white matter. However, the result is shown below when I use all five folds to predict the new scan.

The progress of one of the folds. All folds showed a similar trend. And I used the 100th epoch result to predict. However, most of the cases are shown below:
![image](https://github.com/MIC-DKFZ/nnUNet/assets/56018137/44cce4e8-6700-4c46-954c-5d36a15b9c5d)

Ground truth (left and right white matter)
![image](https://github.com/MIC-DKFZ/nnUNet/assets/56018137/4bcc6506-7bf5-4152-80e6-320b580dd22d)
Prediction
![image](https://github.com/MIC-DKFZ/nnUNet/assets/56018137/13b1b6df-7eaf-436b-979f-08f9e3e5faea)

As you can see, most of the white matter voxels are predicted into the left white matter. I assume there may be a problem with the overfitting.

Now, I have some potential solutions to this:
1. Use the `nnUNetTrainerNoMirroring` Trainer you mentioned, and use the 100th epoch checkpoints to predict.
2. Train less epoch (like maybe the 20th; refer to my progress loss function plot) and use the `nnUNetTrainerNoMirroring` 


I would greatly appreciate it if you and your team could come up with a better solution to address this issue. My hope is that this model can accurately predict the left and right white matter.


Best,

**Comment by lisakausch** [comment]
Hi @chocolatetoast-chiu ,

did you enable `tta` during inference, tta will do mirroring and cause the wrong labeling. `nnUNetTrainerNoMirroring` automatically disables tta.

Best,
Lisa

**Comment by chocolatetoast-chiu** [comment]
Hi @lisakausch ,

Yes! After I change the tta to disable mode, the results are exactly similar to our ground truth segmentation. Thank you! I really appreciate your help!


Best,
